### PR TITLE
github vulnerability parser added

### PR DESCRIPTION
**Pull request by tahir59**
**Note: DefectDojo is now on Python3.5 and Django 2.2.x Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

When submitting a pull request, please make sure you have completed the following checklist:

- [ ] Your code is flake8 compliant 
- [ ] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted)
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [ ] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [ ] Add applicable tests to the unit tests.
- [ ] Add the proper label to categorize your PR 

Current accepted labels for PRs:
- Import Scans (for new scanners/importers)
- enhancement
- feature
- bugfix
- maintenance (a.k.a chores)
- dependencies

### CODE REVIEW

**Comment by valentijnscholten** [comment]
Thanks, can you add some sample reports + units tests (see other parsers for basic unittests).

**Comment by Maffooch** [comment]
Also add it to our [list](https://github.com/DefectDojo/Documentation/blob/master/docs/integrations.rst) of scanners please!

**Comment by valentijnscholten** [comment]
Thanks, can you check/fix the flake8 errors here: https://travis-ci.org/github/DefectDojo/django-DefectDojo/jobs/688690454

**Comment by valentijnscholten** [comment]
@tahir59 Which api / tool are you using to extract the information? The examples look different from what I can find on the github API docs: https://developer.github.com/v3/code-scanning/

**Comment by valentijnscholten** [comment]
Looks like the parser processes output from the Github GraphQL v4: https://developer.github.com/v4/explorer/

**Comment by tahir59** [comment]
*@valentijnscholten *parser naming updated.

below is the github v4 graphql query to fetch data

query getVulnerabilitiesByOwner($owner: String!) {
  search(query: $owner, type: REPOSITORY, first: 100) {
    nodes {
      ... on Repository {
        name
        vulnerabilityAlerts(last: 100) {
          nodes {
            id
            securityVulnerability {
              severity
              package {
                name
              }
              advisory {
                description
                summary
                identifiers {
                  type
                  value
                }
                references {
                  url
                }
              }
            }
          }
        }
      }
    }
  }
}

**Comment by valentijnscholten** [comment]
Thanks. Could you resolve conflichts, and create a PR to update the documentation: https://github.com/DefectDojo/Documentation

File: dojo/unittests/test_github_vulnerability.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        findings = parser.items[0]
+        self.assertEqual(findings.title, "aabbccddeeff1122334401")
+        self.assertEqual(
+            findings.description,
+            "This is a sample description for sample description from Github API.",
+        )
+        self.assertEqual(
+            findings.severity, "Critical"
+        )
```
Comment (Maffooch) [inline dojo/unittests/test_github_vulnerability.py:31]: This fails in our Travis checks
image

File: dojo/fixtures/test_type.json [hunk 2]
```diff
@@ -571,6 +571,13 @@
 		"model": "dojo.test_type",
 		"pk": 173
 	},
+        {
+                "fields": {
+                        "name": "Github Vulnerability Scan"
+                },
+                "model": "dojo.test_type",
+                "pk": 178
+        },
```
Comment (Maffooch) [inline dojo/fixtures/test_type.json:580]: The pk here matches that of BugCrowd on line 609

Comment (tahir59) [reply dojo/fixtures/test_type.json]: fixed

**Comment by Maffooch** [comment]
Travis is really beating you up on this one! You can run the same tests locally to ensure everything passes before fighting with Travis again. Check the readme [here](https://github.com/DefectDojo/django-DefectDojo/blob/dev/DOCKER.md#run-the-tests-with-docker). Also fix the conflicts please.

File: dojo/unittests/test_github_vulnerability.py (hunk 1 above)
Comment (tahir59) [reply dojo/unittests/test_github_vulnerability.py]: fixed

**Comment by valentijnscholten** [comment]
Still would be good to have the documentation updated as requested above

**Comment by tahir59** [comment]
@valentijnscholten doc updated

**Comment by valentijnscholten** [comment]
https://github.com/DefectDojo/Documentation/pull/110/files

**Comment by valentijnscholten** [comment]
@Maffooch are you also happy now? I have approved it now.

**Pull request merged**